app/controllers/ArrivalNotificationController.scala

```diff
@@ -77,23 +79,35 @@ class ArrivalNotificationController @Inject()(
 
                         }
                         .recover {
-                          case _ => InternalServerError
+                          case _ => {
+                            InternalServerError(Json.toJson(ErrorResponseBuilder.failedSavingArrivalNotification))
+                              .as("application/json")
+                          }
```

Comment:
We could pass the exception into the failSavingArrivalNotification to help the frontend indicate is either the mongo has failed or the POST to the hods.

Not necessary but it could be helpful for troubleshooting.